### PR TITLE
Fix process_checkouts to be more sequential

### PR DESCRIPTION
**Pull request by danamansana**
The main change in this branch is that instead of performing a sequence of operations on a list of checkouts by performing the first operation on every checkout, then the second, etc., we fully process the first checkout, then the second, etc.

There are some smaller changes: adding some logging for when we are rejecting a checkout as a duplicate, as well as breaking up some methods into smaller ones.

### CODE REVIEW

File: lib/item_stream_handler.rb [hunk 1]
```diff
@@ -79,14 +80,26 @@ def convert_record_to_checkout(decoded_records)
       .compact
   end
 
+  def dedup_checkout(checkout)
```
Comment (nonword) [inline lib/item_stream_handler.rb]: Could this be `is_duplicate?` or something to indicate it returns a boolean that indicates whether or not the given checkout is a duplicate?

File: lib/item_stream_handler.rb [hunk 2]
```diff
@@ -19,6 +19,7 @@ def add_checkout(checkout)
     # Add checkout to end:
     @checkouts << checkout
 
+    Application.logger.debug "ItemStreamHandler#handle: Added checkout: #{checkout.id} (#{checkout.barcode}) \"#{checkout.title}\""
```
Comment (nonword) [inline lib/item_stream_handler.rb]: Comment prefix is a tad misleading. Should begin "ItemStreamHandler#add_checkout: ..."

**Review by nonword — COMMENTED** [review]
Looks great. Two small requests

File: lib/item_stream_handler.rb (hunk 2 above)
Comment (danamansana) [reply lib/item_stream_handler.rb]: Fixed!

File: lib/item_stream_handler.rb (hunk 1 above)
Comment (danamansana) [reply lib/item_stream_handler.rb]: Changed!

**Review by nonword — APPROVED** [review]

**Pull request merged**